Add support for auto generating openapi docs by the use of validate middleware

murerkinn opened the pull request:
The idea is to generate an OpenAPI spec by using the schemas provided to the `validate` middleware all across the app. To do that I had to expose the schemas passed to the middleware with the handler so that I can access them in a layer of the express stack, which allows me to iterate over all the routes registered in the application.

To demonstrate the script, I created a todo application under examples.

dashersw commented:
Fantastic idea, thank you! We need 100% test coverage, updates to the README, as well as an easy way to generate the spec, instead of c/p'ing an entire generator file. Is that possible?

murerkinn commented:
Sure! I moved the generator function inside the library and exported it from the root. I also added the tests for the new function and revamped the example to use the new function from the src directory.
